customTests/azure_cuda_bandwidth.nhc

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+##This script tests host to device bandwidth and device to host bandwidth using
+##a script recorded within the test that tests host to device and device to 
+##host copy bandwidths between every combination of gpus and numa domains. The
+##test confirms that each GPU is hitting the expected bandwidths to at least
+##one of the numa domains.
+##
+
+#set expected BW set to default value if argument empty
+EXP_CUDA_BW=$1
+if [[ -z "$EXP_CUDA_BW" ]]; then
+	EXP_CUDA_BW=24
+fi
+
+# location of executables, must match setup location
+EXE_DIR=$2
+if [[ -z "$EXE_DIR" ]]; then
+	EXE_DIR=/opt/azurehpc/test/nhc
+fi
+
+#Catch error codes that may be thrown by the executable passed as the first
+#input, and if an error code is tripped throw the second input as a message
+catch_error() {
+	declare -g output=$($1)
+	err_code=$?
+	if [ $err_code -ne 0 ]; then
+		die "\t $2 $err_code" >&2
+		return 1
+	fi
+	return 0
+
+}
+
+function check_cuda_bw()
+{
+	#Count the number of gpu-name nvidia-smi outputs.
+	error_smi="**Fail** nvidia-smi failed with error code"
+	#Lock graphics clocks to 1400 to eliminate any time for the GPUs to boost.
+	#This likely isn't important for performance here, but we will do it anyway
+	#to be safe.
+	lock_clocks="sudo nvidia-smi -lgc 1400"
+	#catch_error "$lock_clocks" "$error_smi"
+	result=$(catch_error "$lock_clocks" "$error_smi")
+    if ! catch_error "$lock_clocks" "$error_smi"; then
+        return 1
+    fi
+
+	#Count the GPUs.
+	gpu_list="timeout 3m nvidia-smi --query-gpu=name --format=csv,noheader"
+    if ! catch_error "$gpu_list" "$error_smi"; then
+        return 1
+    fi
+	ngpus=$(echo "$output" | wc -l)
+
+	#Run the superbench device to host bandwidth test.
```

Comment:
Maybe say that it is running the device to host bandwidth test based on the one in superbench. I'm not sure it's exactly the same any more.

Reply:
Let me address

Reply:
removed mention of superbench